src/main/java/org/folio/consortia/service/impl/PublicationServiceImpl.java

```diff
@@ -69,6 +76,48 @@ public PublicationResponse publishRequest(UUID consortiumId, PublicationRequest
 
     return buildPublicationResponse(createdPublicationEntity.getId());
   }
+
+  @Override
+  public PublicationDetailsResponse getPublicationDetails(UUID consortiumId, UUID publicationId) {
+    log.debug("getById:: Trying to retrieve publicationDetails by consortiumId: {} and publicationId id: {}", consortiumId, publicationId);
+
+    consortiumService.checkConsortiumExistsOrThrow(consortiumId);
+    var publicationStatusEntity = publicationStatusRepository.findById(publicationId)
+      .orElseThrow(() -> new ResourceNotFoundException("publicationId", String.valueOf(publicationId)));
+
+    var publicationTenantRequestEntity = publicationTenantRequestRepository.findByPcState_Id(publicationId, PageRequest.of(0, Integer.MAX_VALUE));
```

Comment:
It is list in response, so I propose to rename to ptrEntities

Reply:
done